src/plot_api/plot_api.js

```diff
@@ -2274,6 +2274,8 @@ Plotly.animate = function(gd, frameOrGroupNameOrFrameList, animationOpts) {
             var newFrame = trans._currentFrame = trans._frameQueue.shift();
 
             if(newFrame) {
+                gd.emit('plotly_animatingframe', newFrame.name);
```

Comment:
Yeah. I have nothing against that since there's not another way to query frames by name short of looking it up yourself in the input data. What other fields do you think are relevant? I see:

`transitionOpts`, `frameOpts`, `frame` (the data itself), `name`.

Maybe all of them? They're just references, so not really any extra cost. Could pass as an object.